src/material/tree/testing/tree-harness.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {
+  MatNestedTreeNodeHarness,
+  MatTreeNodeHarness,
+  TreeHarnessFilters,
+  TreeNodeHarnessFilters
+} from '@angular/material/tree/testing';
+
+/** Harness for interacting with a standard mat-tree in tests. */
+export class MatTreeHarness extends ComponentHarness {
+  /** The selector for the host element of a `MatTableHarness` instance. */
+  static hostSelector = '.mat-tree';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a tree with specific attributes.
+   * @param options Options for narrowing the search
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: TreeHarnessFilters = {}): HarnessPredicate<MatTreeHarness> {
+    return new HarnessPredicate(MatTreeHarness, options);
+  }
+
+  /** Gets all of the nodes in the tree. */
+  async getNodes(filter: TreeNodeHarnessFilters = {}): Promise<MatTreeNodeHarness[]> {
```

Comment:
@jelbourn @andrewseguin thoughts on whether we should have 2 methods or just a single `getNodes` method regardless of the type of node? It feels to me like we should just have 1 method

Reply:
Having these two methods does seem a bit confusing. If we can guarantee that we can operate the same API for both types of node, I think it's fine to combine them under one method.